src/video_canister_package/src/index.ts

```diff
@@ -0,0 +1,243 @@
+import { Identity } from "@dfinity/agent";
+import { Principal } from "@dfinity/principal";
+
+import { getVideoCanisterActor, getWalletCanisterActor, managementPrincipal } from "./common";
+import { MetaInfo } from "./canisters/video_canister/video_canister.did";
+import { IDL } from "@dfinity/candid";
+
+export interface Video{
+  "name": string,
+  "description": string,
+  "version": bigint,
+  "owner": Principal,
+  "videoBuffer": Buffer,
+}
+
+export interface CreationVideo{
+  "name": string,
+  "description": string,
+  "videoBuffer": Buffer,
+}
+
+const spawnPrincipal = Principal.fromText("ryjl3-tyaaa-aaaaa-aaaba-cai"); 
+
+const creationCycles: bigint = BigInt(200_000_000_000);
+
+const chunkSize = 1024;
+
+export async function uploadVideo(identity: Identity, walletId: Principal, video: CreationVideo, cycles: bigint): Promise<Principal>{
+
+  if (cycles < creationCycles){
+    throw Error("Not enough cycles, need at least " + creationCycles + " for video canister creation");
+  }
+  
+  let videoPrincipal = await createNewCanister(identity, walletId, creationCycles);
+  
+  await checkController(identity, walletId, videoPrincipal);
+
+  await depositCycles(identity, walletId, videoPrincipal, cycles - creationCycles);
```

Comment:
vor dem call abchecken ob cycles > creationCycles. oder in der Method auf 0 abchecken. aber gibt hier gerade edge case dass du die Methode aufrufst obwohl Cycles === creationCycles

Reply:
resolved 5c0790b0b8ca08a4275d5b3413e14def9f799197